spec/unit/transaction/additional_resource_generator_spec.rb

```diff
@@ -182,6 +182,22 @@ def find_vertex(graph, type, title)
         'Generator[thing]', 'Whit[completed_thing]')
     end
 
+    it "should tag the sentinel with the tags of the resource" do
+      graph = relationships_after_eval_generating(<<-MANIFEST, 'Generator[thing]')
+        generator { thing:
+          code => 'notify { hello: }',
+          tag  => 'foo',
+        }
+      MANIFEST
+      whit = find_vertex(graph, :whit, "completed_thing")
+      # this matcher relies on Puppet::Util::Tagging#tagged? which accepts a
+      # list of tags and returns true if _at least one_ tag is present. So we
+      # explicitly check both.
+      expect(whit).to be_tagged("thing")
+      expect(whit).to be_tagged("foo")
+      expect(whit).to be_tagged("generator")
```

Comment:
I think these assertions can be reduced to:

```ruby
expect(whit.tags).to be_superset(['thing', 'foo', 'generator'].to_set)
```

Unfortunately no inline `set` notation, so must use `Set.new()` or `[].to_set`